File: algos_semina/3rd_seminar/B_20206/kkddiiock_20206.cpp
from sys import stdin
from collections import deque

dx= [1, -1, 0, 0]
dy= [0, 0, 1, -1]

n, m= map(int, stdin.readline().rstrip().split())
camera= [list(map(int, stdin.readline().rstrip().split())) for _ in range(n)]
vaccine= [list(map(int, stdin.readline().rstrip().split())) for _ in range(n)]
mark=[]
tmp=[]
v=[]

def dfs(a, b, c, d):
    global mark
    global tmp
    if([c, d] not in mark and camera[a][b]==camera[c][d]):
        mark.append([c, d])
        tmp.append([c, d])
        for i in range(4):
            x= c+dx[i]
            y= d+dy[i]
            if(0<=x<n and 0<=y<m):
                dfs(c, d, x, y)
            
def check(n, m):
    global mark
    global tmp
    global v
    for i in range(n):
        for j in range(m):
            if([i, j] not in mark):
                dfs(i, j, i, j)
                v.append(tmp)
                tmp=[]
        
check(n, m)            
cnt=0

for i in range(len(v)):
    a, b= v[i][0][0], v[i][0][1]
    if(camera[a][b]!=vaccine[a][b]):
        cnt+=1

if(cnt==1 or cnt==0):
    print('YES')
else:
    print('NO')

File: algos_semina/3rd_seminar/B_20206/kkddiiock_20206.py
from sys import stdin

a, b, c = list(map(int, stdin.readline().rstrip().split()))
x1, x2, y1, y2 = list(map(int, stdin.readline().rstrip().split()))
X = [x1, x2]
Y = [y1, y2]

def graph(bang):
    for x in X :
        for y in Y :
            rect = a*x+b*y+c
            if rect==0:
                print("Lucky")
                return
            if bang==0 and rect>0 :
                bang=1
            elif bang==0 and rect<0 :
                bang=-1
            if rect*bang<0 :
                print("Poor")
                return
    print("Lucky")

graph(0)
